packages/backend/src/QueryEngine.ts

```diff
@@ -96,12 +119,24 @@ export class QueryEngine<DB> {
         ];
     }
 
+    compile<T>(query: T extends Query<DB, infer TTable> ? T : never): {
```

Comment:
@jimezesinachi we've talked about this: don't move files around, it just makes it more difficult to review your code. Please keep the diff small.